llvm/include/llvm/ProfileData/InstrProf.h

```diff
@@ -869,14 +869,18 @@ struct InstrProfRecord {
   inline uint32_t getNumValueDataForSite(uint32_t ValueKind,
                                          uint32_t Site) const;
 
+  /// Return the array of profiled values at \p Site.
+  inline ArrayRef<InstrProfValueData> getValueForSite(uint32_t ValueKind,
+                                                      uint32_t Site) const;
+
   /// Return the array of profiled values at \p Site.
```

Comment:
The new interface returns an ArrayRef and the deprecated interface allows caller to take ownership of the array.

I wonder if there are existing call sites where explicit ownership is preferred over a reference (e.g., easy to reason about lifecycle, etc).

Reply:
If yes I think we may need both.

Reply:
I wouldn't worry about the lifetime of profile data.  The profile data stays available at least during the entire duration of a pass.